pkg/codegen/dotnet/gen_program.go

```diff
@@ -470,6 +470,9 @@ func (g *generator) usingStatements(program *pcl.Program) programUsings {
 					if csharpinfo, ok := def.Language["csharp"].(CSharpPackageInfo); ok {
 						info = csharpinfo
 					}
+					if info.RootNamespace == "" && r.Schema.PackageReference.Namespace() != "" {
+						info.RootNamespace = r.Schema.PackageReference.Namespace()
+					}
```

Comment:
So are we missing a test in pulumi/pulumi? Something failing in pulumi-dotnet sounds like too late?

Reply:
Maybe? This is very awkward to work with tbh.  We've been trying to move more codegen testing to conformance tests, so I didn't want to add a specific codegen test just for this since conformance tests are covering this and better. However dotnet codegen lives in `pulumi/pulumi` and the conformance tests in `pulumi/pulumi-dotnet`, which is awkward.  

I've added an item to the ideation agenda later to talk about conformance tests and maybe making this better. (In my ideal world we'd just have a monorepo with all the languages, but there's some strong opinions against that :))

Reply:
We need to go one way or the other. Either bring everything back into pulumi/pulumi as a mono-repo or move the language specific parts (which is just dotnet codegen now) out to the language repos.

I think we can just move dotnet codegen, we just need to update tfbridge to refer to the new place.